src/database/utils.js

```diff
@@ -124,16 +124,36 @@ function getParsedFilePathObj(filePathObj) {
   return parsedFilePathObjList;
 }
 //Create a new database analysis entry
-export async function getAllAnalysis(event) {
-  return new Promise(resolve => {
-    var getAllAnalysis = db.analysis.find({});
-    resolve(analysisId.$loki);
+export function getAllAnalysis(event, db) {
+  var allAnalysis = db.analysis.find({$loki: {$ne: null}});
+  var allAnalysisIDs = allAnalysis.map(analysis => {
+    return {analysisID: analysis.$loki};
   });
+
+  var allRelations = db.relations.find({$or: allAnalysisIDs});
+  var allFileIDs = allRelations
+    .map(relation => {
+      return {$loki: relation.fileID};
+    })
+    .reduce((finalList, curr) => {
+      finalList =
```

Comment:
this can just be a return of the ternary statement

`(finalList, curr) => finalList.findIndex(...) ? finalList : [...finalList, curr]`